components/director/internal/domain/application/service.go

```diff
@@ -18,7 +18,7 @@ import (
 )
 
 const (
-	intSysKey = "integration-system-id"
+	intSysKey = "integration_system_id"
```

Comment:
Just minor, don't we want to have a convention that underscores is the separator? In this case, I would go with `integrationSystemId`